bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableConfiguration.java

```diff
@@ -87,13 +88,35 @@ public T withConfiguration(String key, String value) {
     }
 
     /**
+     * Used for creating a new {@link Builder} from a {@link CloudBigtableConfiguration} or its
+     * subclasses.
+     */
+    protected T setConfiguration(Map<String, String> configuration) {
+      this.additionalConfiguration.putAll(configuration);
```

Comment:
how about this?
this.additionalConfiguration.putAll(configuration);
this.projectId = additionalConfiguration.remove(BigtableOptionsFactory.PROJECT_ID_KEY);
this.zoneId = additionalConfiguration.remove(BigtableOptionsFactory.ZONE_KEY);
this.clusterId = additionalConfiguration.remove(BigtableOptionsFactory.CLUSTER_KEY);